### ISSUE TITLE
First-Install broken - missing mkdir in working folder

### ISSUE THREAD

**Issue by GitMensch** [spgennard/vscode_cobol#242]
**Describe the bug**
Extension cannot be activated.
This only happens if the user never has used the extension before.

**To Reproduce**
1. Open Editor
2. check the COBOL output for the storagepath
3. Open that in Windows explorer
4. Close Editor
5. Rename storagepath
6. Start editor
7. See error (extension does not load, the Output for the Extension Host Protocol shows the reason):

```
[2020-11-11 13:30:35.971] [exthost] [error] Activating extension bitlang.cobol failed due to an error:
[2020-11-11 13:30:35.974] [exthost] [error] Error: ENOENT: no such file or directory, mkdir 'c:\Users\USER\AppData\Roaming\VSCodium\User\workspaceStorage\9e0abf2c922a5ecd86d9d585469903c9\bitlang.cobol\USER'
	at mkdirSync (fs.js:823:3)
	at Object.fs.mkdirSync (electron/js2c/asar.js:718:16)
	at Function.getStoragePathArea (c:\Users\USER\.vscode-oss\extensions\bitlang.cobol-6.10.3\dist\extension.js:2:200853)
	at Function.getCacheDirectory (c:\Users\USER\.vscode-oss\extensions\bitlang.cobol-6.10.3\dist\extension.js:2:201134)
	at ie (c:\Users\USER\.vscode-oss\extensions\bitlang.cobol-6.10.3\dist\extension.js:2:114211)
	at t.activate (c:\Users\USER\.vscode-oss\extensions\bitlang.cobol-6.10.3\dist\extension.js:2:116982)
	at Function._callActivateOptional (d:\VSCodium\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:933:980)
	at Function._callActivate (d:\VSCodium\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:933:631)
	at d:\VSCodium\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:932:126
	at processTicksAndRejections (internal/process/task_queues.js:94:5)
	at async Promise.all (index 0)
```

**Expected behavior**
Let the extension work out of the box. In Unix-speech:

```sh
mkdir 'c:\Users\USER\AppData\Roaming\VSCodium\User\workspaceStorage\9e0abf2c922a5ecd86d9d585469903c9\bitlang.cobol\USER' # bad
mkdir -p 'c:\Users\USER\AppData\Roaming\VSCodium\User\workspaceStorage\9e0abf2c922a5ecd86d9d585469903c9\bitlang.cobol\USER' # good
```

**Workaround**
Manually create the necessary `bitlang.cobol` folder.

**Comment by GitMensch** [comment]
Note: it obviously makes no sense to have the user and the workspace id at all in the workspaceStorage. The reason: each workspace has a different ID = so different folder and is stored in the users home data directory = different per user.

for `"coboleditor.cache_metadata": "storagepath"`: please drop the additional sub-folders. In the case above, after manually creating the bitlang.cobol folder, the storagepath evaluates to

'c:\Users\USER\AppData\Roaming\VSCodium\User\workspaceStorage\9e0abf2c922a5ecd86d9d585469903c9\bitlang.cobol\USER\f00b06fce209b941ff2b9a6714a46820' - but as the initial `9e0abf2c922a5ecd86d9d585469903c9`, which is created by vscode, is already unique per workspace and the username in front is also unique the plain 'c:\Users\USER\AppData\Roaming\VSCodium\User\workspaceStorage\9e0abf2c922a5ecd86d9d585469903c9\bitlang.cobol` would be better (I _think_ this is what previous versions used).

**Comment by spgennard** [comment]
Sorry, username is required for shared storage when used with multi users web based vscode installs.

I am seriously thinking about removing the storagepath support as I received complaints about it.

**Comment by GitMensch** [comment]
> Sorry, username is required for shared storage when used with multi users web based vscode installs.

I see. So that's completely reasonable to stay then and the only missing part is the `mkdir -p` (creation of parent directories).

> I am seriously thinking about removing the storagepath support as I received complaints about it.

Please don't, that's the only way for our programmers to work with the extension at all - they can't write to the workspace path, only to single files that are checked-out before.

**Comment by spgennard** [comment]
The complaint I get is the "creeping" size of the storage area.

**Comment by GitMensch** [comment]
That's strange, but in any case it is an optional non-default setting, so what are people complaining about?

> removing my source code" does not remove the cache

I see, that's something a complete metadata parsing either handles already or could do relative easily.

**Comment by spgennard** [comment]
Code is present to create the user/id area, so it must the storage area itself that is missing.

```javascript
        const storagePathPlusUser = path.join(storagePath,VSCOBOLSourceScanner.username);
        if (!fs.existsSync(storagePathPlusUser)) {
            fs.mkdirSync(storagePathPlusUser);
        }
        const completePath = path.join(storagePathPlusUser, storageid);
        if (!fs.existsSync(completePath)) {
            fs.mkdirSync(completePath);
        }
```

**Comment by spgennard** [comment]
> I see, that's something a complete metadata parsing either handles already or could do relative easily.

No because the area is "orphaned" from the workspace config.. ie: nothing references it

**Comment by GitMensch** [comment]
> Code is present to create the user/id area, so it must the storage area itself that is missing.

No, as outlined above the "bitlang.cobol" part is missing.

**Comment by spgennard** [comment]
Issue resolved with work on #243
